src/jdk.incubator.vector/share/classes/jdk/incubator/vector/Vector.java

```diff
@@ -2766,6 +2766,29 @@ public abstract VectorMask<E> compare(VectorOperators.Comparison op,
      */
     public abstract Vector<E> selectFrom(Vector<E> v);
 
+    /**
+     * Rearranges the lane elements of two vectors, selecting lanes
```

Comment:
We already have another flavor of [selectFrom](https://docs.oracle.com/en/java/javase/22/docs/api/jdk.incubator.vector/jdk/incubator/vector/Vector.html#selectFrom(jdk.incubator.vector.Vector)) which permutes single vector, new API extents its semantics to two vector selection, so we kept the nomenclature consistent.

Reply:
Select operates only on vectors where the `this` vector represents the indexes to *select* elements from the other vectors. Rearrange operates on vectors and a shuffle argument that *rearranges* elements from the other vectors. The former behavior can be specified in terms of the latter behavior, and ideally the equivalent expressions should result in ~same generated sequence of instructions. However, we are not there yet and need to further optimize shuffles to make that happen. But, we can optimize `selectFrom` with the dependent change to wrap indexes instead of throwing when out of bounds.

(Separately there is an annoying issue with select, that we should not address in this PR. Using a Float/Double Vector for indexes is awkward.)